core/src/main/java/org/opensearch/sql/calcite/CalciteRelNodeVisitor.java

```diff
@@ -174,6 +175,31 @@ public RelNode visitFilter(Filter node, CalcitePlanContext context) {
     return context.relBuilder.peek();
   }
 
+  @Override
+  public RelNode visitRegex(Regex node, CalcitePlanContext context) {
+    visitChildren(node, context);
+
+    // TODO: Handle the default field case, tracking by
+    // https://github.com/opensearch-project/sql/issues/4111
+    if (node.getField() == null) {
+      throw new IllegalArgumentException("Regex command requires a field to be specified");
+    }
```

Comment:
Is this possible? Field is not optional in grammar right?

Reply:
This is just for now. I do have an issue to track the default field implementation for both `regex` and `rex` : https://github.com/opensearch-project/sql/issues/4111

Reply:
Oh, i saw the issue, this is not being triggered at all.

Reply:
removed